store/sqlstore/post_store.go

```diff
@@ -2529,11 +2533,22 @@ func (s *SqlPostStore) SearchPostsForUser(paramsList []*model.SearchParams, user
 		return nil, err
 	}
 
-	var wg sync.WaitGroup
-
+	now := model.GetMillis()
 	pchan := make(chan store.StoreResult, len(paramsList))
 
+	var wg sync.WaitGroup
 	for _, params := range paramsList {
+		// Deliberately keeping non-alphanumeric characters to
+		// prevent surprises in UI.
+		buf, err := json.Marshal(params)
+		if err != nil {
+			return nil, err
+		}
+		err = s.LogRecentSearch(userID, buf, now)
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
Looks like this logic iterates through each parameter passed in the API and then stores it as Recent Search. Shouldn't the logic be once per API?

Reply:
Discussed offline. ParamsList is a list of queries, and each item contains the struct containing all the params.